nominatim/index.c

```diff
@@ -283,6 +286,23 @@ void nominatim_index(int rank_min, int rank_max, int num_threads, const char *co
         exit(EXIT_FAILURE);
     }
 
+    res = PQexec(conn, "SHOW lc_messages");
+    status_code = PQresultStatus(res);
+    if (status_code != PGRES_TUPLES_OK && status_code != PGRES_SINGLE_TUPLE) {
+        fprintf(stderr, "Failed determining database locale: %s\n", PQerrorMessage(conn));
+        exit(EXIT_FAILURE);
+    }
+    num_rows = PQntuples(res);
+    if (num_rows > 0)
+    {
+        result_string = PQgetvalue(res, 0, 0);
+        if (result_string && (strlen(result_string) > 0) && (strcasecmp(result_string, "C") != 0))
+        {
+            // non-default locale if the result exists, is non-empty, and is not "C"
+            db_has_locale = 1;
+        }
+    }
```

Comment:
Depending on the configuration of the Postgres cluster, a remote user may not be allowed to run the `SET lc_messages to 'C'` command. Since 'C' is the default locale setting (or, rather, lack thereof) on a Postgres cluster, it is often unnecessary for this setup script to change it. This change guards against that case, and doesn't try to change `LC_MESSAGES` to 'C' if that is already the configuration.

Reply:
The code later detects deadlocks by parsing the error message 'ERROR:  deadlock detected'. If that message doesn't come out in English, that part of the code will fail. I wonder if it would be better to error out at this point if the locale is not in English and cannot be set. Or if you know of a less brittle way to detect the error condition, that would be even better.

Reply:
I hadn't noticed that deadlock detection logic. I don't know of a more reliable way to ascertain whether a failed query was the result of the server's built in deadlock detection algorithm. However, my change still supports the existing logic: in essence, the new logic is "Set LC_MESSAGES to 'C' **only if it isn't already**".